src/ResourceManagement/CosmosDB/IndexingPolicyImpl.cs

```diff
@@ -0,0 +1,278 @@
+{
+
+    internal partial class IndexingPolicyImpl<ParentImplT, IParentT, DefinitionParentT, UpdateParentT> :
+        ChildResource<Models.IndexingPolicy, ParentImplT, IParentT>,
+        IndexingPolicy.Definition.IDefinition<DefinitionParentT>,
+        IndexingPolicy.Update.IUpdate<UpdateParentT>
+        where ParentImplT : IParentT, DefinitionParentT, UpdateParentT, IHasIndexingPolicy<ParentImplT>
+    {
+        public IndexingPolicyImpl(Models.IndexingPolicy inner, ParentImplT parent)
+            : base(inner, parent)
+        {
+        }
+
+        public override string Name()
+        {
+            return null;
+        }
+
+        public ParentImplT Attach()
+        {
+            return this.Parent.WithIndexingPolicy(this.Inner);
+        }
+
+        public IndexingPolicyImpl<ParentImplT, IParentT, DefinitionParentT, UpdateParentT> WithAutomatic(bool isAutomatic)
+        {
+            Inner.Automatic = isAutomatic;
+            return this;
+        }
+
+        public IndexingPolicyImpl<ParentImplT, IParentT, DefinitionParentT, UpdateParentT> WithIndexingMode(IndexingMode indexingMode)
+        {
+            Inner.IndexingMode = indexingMode;
+            return this;
+        }
+
+        public IndexingPolicyImpl<ParentImplT, IParentT, DefinitionParentT, UpdateParentT> WithoutIndexingMode()
+        {
+            Inner.IndexingMode = null;
+            return this;
+        }
+
+        public IndexingPolicyImpl<ParentImplT, IParentT, DefinitionParentT, UpdateParentT> WithIncludedPathsAppend(IList<IncludedPath> includedPaths)
+        {
+            if (Inner.IncludedPaths == null)
+            {
+                Inner.IncludedPaths = new List<IncludedPath>();
+            }
+
+            foreach (var includePath in includedPaths)
+            {
+                Inner.IncludedPaths.Add(includePath);
+            }
+            return this;
+        }
+
+        public IndexingPolicyImpl<ParentImplT, IParentT, DefinitionParentT, UpdateParentT> WithIncludedPathsReplace(IList<IncludedPath> includedPaths)
+        {
+            Inner.IncludedPaths = includedPaths;
+            return this;
+        }
+
+        public IndexingPolicyImpl<ParentImplT, IParentT, DefinitionParentT, UpdateParentT> WithIncludedPath(int index, IncludedPath includedPath)
+        {
+            if (Inner.IncludedPaths == null)
+            {
+                Inner.IncludedPaths = new List<IncludedPath>();
+            }
+
+            if (index < 0 || Inner.IncludedPaths.Count <= index)
+            {
+                Inner.IncludedPaths.Add(includedPath);
+            }
+            else
+            {
+                Inner.IncludedPaths[index] = includedPath;
+            }
+
+            return this;
+        }
+
+        public IndexingPolicyImpl<ParentImplT, IParentT, DefinitionParentT, UpdateParentT> WithoutIncludedPath(int index)
+        {
+            Inner.IncludedPaths.RemoveAt(index);
+            return this;
+        }
```

Comment:
Not sure whether we actually need the indexed operations on list. Seems not easy to use anyway. Hard to remove anyway.

However we might support `WithIncludedPathsAppend(IncludedPath includedPaths)` to add one line, which can be called multiple times.

@yaohaizh

Reply:
I have removed all function about index in list, just use the type inside list.